### ISSUE TITLE
Visual Studio projects generated by CMake use /O2 instead of /Od on Debug configuration

### ISSUE THREAD

**Issue by slomp** [halide/Halide#7057]
I think the title is pretty self-explanatory, but let me know if more details are needed.

This is quite strange, since my experience with CMake when generating Visual Studio projects always produced /Od for the Debug configuration.

**Comment by alexreinking** [comment]
Yeah, we mess around with flags way too much in this build. I'll take a look...

**Comment by slomp** [comment]
Also (probably deserves a new ticket): I can't generate Visual Studio projects using the Linux environment on Windows (including MSYS). It fails to find LLVM-Config.txt, but maybe it's vcpkg's fault, and not ours.

**Comment by alexreinking** [comment]
@slomp - I don't think we can do anything to address that. AFAIK, you cannot run the Visual Studio generator from WSL, period. You have to run it from Windows in either PowerShell or cmd.

**Comment by slomp** [comment]
Yeah, unfortunately that also prevent us from running `run-clang-tidy.sh`, since it seems to require calling cmake.

**Comment by steven-johnson** [comment]
> Yeah, unfortunately that also prevent us from running `run-clang-tidy.sh`, since it seems to require calling cmake.

clang-tidy is *always* supposed to use a captive version of Clang (never MSVC), so that shouldn't matter.

(I can believe that the script might not work under WSL/msys/etc for any other number of reasons, of course)
